source/common/network/connection_impl.cc

```diff
@@ -744,8 +743,8 @@ void ConnectionImpl::onWriteReady() {
   if (connecting_) {
     int error;
     socklen_t error_size = sizeof(error);
-    RELEASE_ASSERT(
-        socket_->getSocketOption(SOL_SOCKET, SO_ERROR, &error, &error_size).return_value_ == 0, "");
+    auto rc = socket_->getSocketOption(SOL_SOCKET, SO_ERROR, &error, &error_size).return_value_;
```

Comment:
I think the `error` should be initialized to -1 above, so it will take the error path below. Otherwise the outcome of this function is random.